File: Data Structures & Algorithms/subarray-sum-equals-k/submission-2.rs
use std::collections::HashMap;
impl Solution {
    pub fn subarray_sum(nums: Vec<i32>, k: i32) -> i32 {
        let n: usize = nums.len();
        let mut count: i32 = 0;
        let mut psum: i32 = 0;
        let mut store: HashMap<i32, i32> = HashMap::new();
        store.insert(0,1);
        for num in &nums {
            psum += num;
            let diff: i32 = psum - k;
            if let Some(&freq) = store.get(&diff) {
                count += freq;
            }
            *store.entry(psum).or_insert(0) += 1;
        }
        count
    }
}

File: Data Structures & Algorithms/rotate-array/submission-2.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        n = len(nums)
        nums.reverse()
        k = k % n
        def r(l,r):
            while l<r:
                nums[l], nums[r] = nums[r], nums[l]
                l += 1
                r -= 1
        
        r(0, k-1)
        r(k, n-1)
